### PR TITLE
Do not free the uninitialized cstring.

### PR DESCRIPTION
**Pull request by lonicerae**
* Better deal with corrupted queue messages

### CODE REVIEW

File: runtime/stringbuf.c [hunk 1]
```diff
@@ -219,7 +219,7 @@ rsCStrConstructFromCStr(cstr_t **const ppThis, const cstr_t *const pFrom)
 
 void rsCStrDestruct(cstr_t **const ppThis)
 {
-	free((*ppThis)->pBuf);
+	if ((*ppThis)->pBuf) free((*ppThis)->pBuf);
```
Comment (Cropi) [inline runtime/stringbuf.c]: I think there is no need to check the condition. If `(*ppThis)->pBuf` is `NULL`, then no operation is performed by the `free()` function as per `man 3 free`.

**Comment by Cropi** [comment]
I do confirm that the patch fixes the issue. Another approach could be to change `step <= 2` to `step <= 3` in `objDeserializeProperty`:
```
if(iRet != RS_RET_OK && iRet != RS_RET_NO_PROPLINE) {
  if(step <= 2) {
	  pProp->varType = VARTYPE_NONE;
  }
}
```

One thing that we should also investigate is how the queue got malformed. Is there a possibility that there were 2 concurrent writes to the same queue file?

File: runtime/stringbuf.c (hunk 1 above)
Comment (lonicerae) [reply runtime/stringbuf.c]: Ah, yes. Let's keep it unchanged.

**Review by Cropi — APPROVED** [review]

**Comment by Cropi** [comment]
Can we add this to the next milestone? Thanks.

**Comment by rgerhards** [comment]
> Can we add this to the next milestone? Thanks. @rgerhards

thx, I seem to have overlooked this

**Pull request merged**